google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/TestBench.java

```diff
@@ -0,0 +1,307 @@
+  private final String baseUri;
+  private final String dockerImageName;
+  private final String dockerImageTag;
+
+  private final Gson gson;
+  private final HttpRequestFactory requestFactory;
+
+  private TestBench(
+      boolean ignorePullError, String baseUri, String dockerImageName, String dockerImageTag) {
+    this.ignorePullError = ignorePullError;
+    this.baseUri = baseUri;
+    this.dockerImageName = dockerImageName;
+    this.dockerImageTag = dockerImageTag;
+    this.gson = new Gson();
+    this.requestFactory = new NetHttpTransport.Builder().build().createRequestFactory();
+  }
+
+  String getBaseUri() {
+    return baseUri;
+  }
+
+  RetryTestResource createRetryTest(RetryTestResource retryTestResource) throws IOException {
+    GenericUrl url = new GenericUrl(baseUri + "/retry_test");
+    String jsonString = gson.toJson(retryTestResource);
+    HttpContent content =
+        new ByteArrayContent("application/json", jsonString.getBytes(StandardCharsets.UTF_8));
+    HttpRequest req = requestFactory.buildPostRequest(url, content);
+    setCommonReqProperties(req);
+    HttpResponse resp = req.execute();
+    RetryTestResource result = gson.fromJson(resp.parseAsString(), RetryTestResource.class);
+    resp.disconnect();
+    return result;
+  }
+
+  void deleteRetryTest(RetryTestResource retryTestResource) throws IOException {
+    GenericUrl url = new GenericUrl(baseUri + "/retry_test/" + retryTestResource.id);
+    HttpRequest req = requestFactory.buildDeleteRequest(url);
+    setCommonReqProperties(req);
+    HttpResponse resp = req.execute();
+    resp.disconnect();
+  }
+
+  RetryTestResource getRetryTest(RetryTestResource retryTestResource) throws IOException {
+    GenericUrl url = new GenericUrl(baseUri + "/retry_test/" + retryTestResource.id);
+    HttpRequest req = requestFactory.buildGetRequest(url);
+    HttpResponse resp = req.execute();
+    setCommonReqProperties(req);
+    RetryTestResource result = gson.fromJson(resp.parseAsString(), RetryTestResource.class);
+    resp.disconnect();
+    return result;
+  }
+
+  List<RetryTestResource> listRetryTests() throws IOException {
+    GenericUrl url = new GenericUrl(baseUri + "/retry_tests");
+    HttpRequest req = requestFactory.buildGetRequest(url);
+    setCommonReqProperties(req);
+    HttpResponse resp = req.execute();
+    JsonObject result = gson.fromJson(resp.parseAsString(), JsonObject.class);
+    JsonArray retryTest = (JsonArray) result.get("retry_test");
+    ImmutableList.Builder<RetryTestResource> b = ImmutableList.builder();
+    for (JsonElement e : retryTest) {
+      b.add(gson.fromJson(e, RetryTestResource.class));
+    }
+    resp.disconnect();
+    return b.build();
+  }
+
+  @Override
+  public Statement apply(final Statement base, Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        Path tempDirectory = Files.createTempDirectory("retry-conformance-server");
```

Comment:
What happens with this file? Is it saved for debugging or just redirected to something that will be cleaned up?

Reply:
This directly will contain two temp files `stdout` and `stderr` containing the redirected output from the testbench server.